src/Controller/Component/RequestHandlerComponent.php

```diff
@@ -259,13 +259,19 @@ public function beforeRedirect(Event $event, $url, Response $response)
         if (is_array($url)) {
             $url = Router::url($url + ['_base' => false]);
         }
+        $query = [];
+        if (strpos($url, '?') !== false) {
+            list($url, $querystr) = explode('?', $url);
```

Comment:
Not really sure about this... If $url passed to beforeRedirect would already contain an invalid querystring with multiple "?", e.g.
    `$url = "xyz?a=1&b=2?c=3"`
now it results after explode in:
    `$url = "xyz"; $querystr = "a=1&b=2"`
which is right. With limit 2 for explode this would result in a still invalid $querystr:
   `$url = "xyz"; $querystr = "a=1&b=2?c=3"`